tests/Unit/PrestaShopBundle/Translation/TranslationCatalogueBuilderTest.php

```diff
@@ -0,0 +1,482 @@
+        $router->method('generate')->willReturn('route');
+
+        $this->translationCatalogueBuilder = new TranslationCatalogueBuilder($providerFactory, new TranslationsTreeBuilder($router));
+    }
+
+    public function testGetDomainCatalogueWrongType()
+    {
+        $this->expectException(UnexpectedTranslationTypeException::class);
+        $this->translationCatalogueBuilder->getDomainCatalogue(
+            'toto',
+            'en',
+            'domain',
+            [],
+            'theme',
+            'module'
+        );
+    }
+
+    public function testGetDomainCatalogueEmptyTheme()
+    {
+        $this->expectException(\Exception::class);
+        $this->translationCatalogueBuilder->getDomainCatalogue(
+            TranslationCatalogueBuilder::TYPE_THEMES,
+            'en',
+            'domain',
+            [],
+            '',
+            'module'
+        );
+    }
+
+    public function testGetDomainCatalogueEmptyModule()
+    {
+        $this->expectException(\Exception::class);
+        $this->translationCatalogueBuilder->getDomainCatalogue(
+            TranslationCatalogueBuilder::TYPE_MODULES,
+            'en',
+            'domain',
+            [],
+            'theme',
+            ''
+        );
+    }
+
+    public function testGetDomainCatalogueStructure()
+    {
+        $catalogue = $this->translationCatalogueBuilder->getDomainCatalogue(
+            TranslationCatalogueBuilder::TYPE_BACK,
+            'en',
+            'AdminSecondDomain',
+            [],
+            'theme',
+            'module'
+        );
+
+        $this->assertArrayHasKey('data', $catalogue);
+        $this->assertArrayHasKey('info', $catalogue);
+
+        $this->assertCount(2, $catalogue['data']);
+
+        $this->assertSame([
+            'default' => 'Second Domain Second Wording',
+            'xliff' => null,
+            'database' => null,
+            'tree_domain' => [
+                'Admin',
+                'Second',
+                'Domain',
+            ],
+        ], $catalogue['data'][0]);
+
+        $this->assertSame([
+            'default' => 'Second Domain First Wording',
+            'xliff' => 'Second Domain First Wording File Translation',
+            'database' => null,
+            'tree_domain' => [
+                'Admin',
+                'Second',
+                'Domain',
+            ],
+        ], $catalogue['data'][1]);
+
+        $this->assertArrayHasKey('locale', $catalogue['info']);
+        $this->assertArrayHasKey('domain', $catalogue['info']);
+        $this->assertArrayHasKey('theme', $catalogue['info']);
+        $this->assertArrayHasKey('total_translations', $catalogue['info']);
+        $this->assertSame(2, $catalogue['info']['total_translations']);
+        $this->assertArrayHasKey('total_missing_translations', $catalogue['info']);
+        $this->assertSame(1, $catalogue['info']['total_missing_translations']);
+    }
+
+    public function testGetDomainCatalogueSearch()
```

Comment:
```suggestion
    public function testGetTheDomainCatalogueWithSearchTerms()
```

Don't hesitate to state what you expect 😉 
```
/**
 * This should return the catalog BUT only translations matching the search should be
 * part of the catalog
 */
```